fuel-merkle/src/sparse/merkle_tree.rs

```diff
@@ -524,14 +521,27 @@ where
         }
 
         // Merge side nodes
-        for side_node in side_nodes {
-            current_node = Node::create_node_on_path(path, &current_node, side_node);
+        for (side_node, old_parent) in
+            side_nodes.iter().zip(path_nodes.iter().skip(1 /* leaf */))
```

Comment:
nit: Moving `path_nodes.iter().skip(1)` to a separate line with a let binding might make this more readable.